frontend/src/lib/components/SocialLoginButton/index.tsx

```diff
@@ -2,52 +2,70 @@ import { useValues } from 'kea'
 import './index.scss'
 import clsx from 'clsx'
 import { SocialLoginIcon } from './SocialLoginIcon'
-import { SSOProviders } from '~/types'
+import { SSOProvider } from '~/types'
 import { SSO_PROVIDER_NAMES } from 'lib/constants'
 import { preflightLogic } from 'scenes/PreflightCheck/preflightLogic'
 import { LemonButton } from 'lib/lemon-ui/LemonButton'
 import { LemonDivider } from 'lib/lemon-ui/LemonDivider'
-import { router } from 'kea-router'
+import { router, combineUrl } from 'kea-router'
 
-interface SharedProps {
-    queryString?: string
+interface SocialLoginLinkProps {
+    provider: SSOProvider
+    extraQueryParams?: Record<string, string>
+    children: JSX.Element
 }
 
-interface SocialLoginButtonProps extends SharedProps {
-    provider: SSOProviders
+export function SocialLoginLink({ provider, extraQueryParams, children }: SocialLoginLinkProps): JSX.Element {
+    const { searchParams } = useValues(router)
+
+    const loginParams: Record<string, string> = { ...extraQueryParams }
+    if (searchParams.next) {
+        loginParams.next = searchParams.next
+    }
+    if (provider === 'saml') {
+        // SAML-based login requires an extra param as technically we can support multiple SAML backends
+        loginParams.idp = 'posthog_custom'
+    }
+    const loginUrl = combineUrl(`/login/${provider}`, loginParams).url
+
+    return (
+        <a className="block" href={loginUrl}>
+            {children}
+        </a>
+    )
 }
 
-interface SocialLoginButtonsProps extends SharedProps {
-    title?: string
-    caption?: string
-    captionLocation?: 'top' | 'bottom'
-    className?: string
-    topDivider?: boolean
-    bottomDivider?: boolean
+interface SocialLoginButtonProps {
+    provider: SSOProvider
+    redirectQueryParams?: Record<string, string>
 }
 
-export function SocialLoginLink({ provider, queryString }: SocialLoginButtonProps): JSX.Element | null {
+export function SocialLoginButton({ provider, redirectQueryParams }: SocialLoginButtonProps): JSX.Element | null {
     const { preflight } = useValues(preflightLogic)
 
     if (!preflight?.available_social_auth_providers[provider]) {
         return null
     }
 
-    // SAML-based login requires an extra param as technically we can support multiple SAML backends
-    const extraParam = provider === 'saml' ? (queryString ? '&idp=posthog_custom' : '?idp=posthog_custom') : ''
-
     return (
-        <LemonButton
-            size="medium"
-            to={`/login/${provider}/${queryString || ''}${extraParam}`}
-            disableClientSideRouting
```

Comment:
I guess that shifting from `Link` with `disableClientSideRouting` to an `a` is the cause here https://posthog.slack.com/archives/C0185UNBSJZ/p1678302859353109

Reply:
A `<Link disableClientSideRouting to="">` is literally an `<a href="">`, except with less React complexity involved, so this change is a no-op. Turns out it's the lack of a trailing slash!